dashboard/app/models/stage.rb

```diff
@@ -75,6 +75,32 @@ def summarize
         levels: script.script_levels.to_a.select{|sl| sl.stage_id == id}.map(&:summarize),
     }
 
+    # Go through all levels.  If we find an assessment with a per_page value,
+    # then it's a long assessment: we assume it's the final level in the
+    # stage, and generate some placeholder levels for the subsequent pages
+    # of the long assessment.  Each of those levels will have the same basic
+    # URL as the first, but with ?page=1, ?page=2, etc.
+
+    levels = stage_data[:levels]
+
+    # We might append additional levels, so iterate over the original set.
+    levels[0..levels.length].each do |level|
+      if level[:kind] == "assessment"
```

Comment:
Got a thought on how to do this?  I deliberately kept it away from the model because I didn't want any of the systems that rely on the actual models to know about these extra levels, and am deliberately generating them in the summarize family of functions so that they just magically appear closer to the progress-rendering.